src/Plugin.php

```diff
@@ -110,12 +116,26 @@ private function hook() {
 		add_filter( 'action_scheduler_store_class', [ $this, 'set_store_class' ], 10, 1 );
 		add_filter( 'action_scheduler_logger_class', [ $this, 'set_logger_class' ], 10, 1 );
 		add_action( 'plugins_loaded', [ $this, 'register_cli_command' ], 10, 0 );
+		add_action( 'init', [ $this, 'maybe_hook_migration' ] );
 		add_action( 'shutdown', [ $this, 'schedule_migration' ], 0, 0 );
 
 		// TODO: add this to the correct load-* hook once the admin page exists
 		add_action( 'load-action-scheduler', [ $this, 'hook_admin_notices' ], 10, 0 );
 	}
 
+	/**
+	 * Possibly hook the migration scheduler action.
+	 *
+	 * @author Jeremy Pry
+	 */
+	public function maybe_hook_migration() {
+		if ( $this->migration_scheduler->is_migration_complete() ) {
```

Comment:
We could check whether the migration is scheduled, but if it's not actually scheduled then the hook won't fire, so I don't think there's any harm. One upside of *not* checking `is_migration_scheduled()` is that we avoid the queries to the store to check for the migration action.

I think it's fine to leave off the check, but if you think we need to include it, I don't think it would be bad to do so.